src/registrar/assets/js/get-gov.js

```diff
@@ -512,16 +519,24 @@ function hideDeletedForms() {
       formNum++;
 
       newForm.innerHTML = newForm.innerHTML.replace(formNumberRegex, `${formIdentifier}-${formNum-1}-`);
-      // For the other contacts form, we need to update the fieldset headers based on what's visible vs hidden,
-      // since the form on the backend employs Django's DELETE widget. For the other formsets, we delete the form
-      // in JS (completely remove from teh DOM) so we update the headers/labels based on total number of forms.
       if (isOtherContactsForm) {
+        // For the other contacts form, we need to update the fieldset headers based on what's visible vs hidden,
+        // since the form on the backend employs Django's DELETE widget.
         let totalShownForms = document.querySelectorAll(`.repeatable-form:not([style*="display: none"])`).length;
         newForm.innerHTML = newForm.innerHTML.replace(formLabelRegex, `${formLabel} ${totalShownForms + 1}`);
       } else {
-        newForm.innerHTML = newForm.innerHTML.replace(formLabelRegex, `${formLabel} ${formNum}`);
+        // Nameservers form is cloned from index 2 which has the word optional on init, does not have the word optional
+        // if indices 0 or 1 have been deleted
+        let containsOptional = newForm.innerHTML.includes('(optional)');
+        if (isNameserversForm && !containsOptional) {
+          newForm.innerHTML = newForm.innerHTML.replace(formLabelRegex, `${formLabel} ${formNum} (optional)`);
+        } else {
+          newForm.innerHTML = newForm.innerHTML.replace(formLabelRegex, `${formLabel} ${formNum}`);
+        }
```

Comment:
(Optional) I think there may an existing override on the backend (get_form?) that you may be able to use to make some of these checks a bit easier. Rather than check for the word optional, consider adding some sort of toggleable tag (like isOptional or even a class) to the html content on init (or within the JS init).

Then you can add a function which checks for this tag and does some work on the formLabel if its found (plus maybe toggling the bool). For example,
```
if form.isOptional:
newForm.innerHTML = newForm.innerHTML.replace(formLabelRegex, `${formLabel} ${formNum} (optional)`);
else: 
newForm.innerHTML = newForm.innerHTML.replace(formLabelRegex, `${formLabel} ${formNum}`);
```
That way, this 'function' could be reused in both areas and it simplifies the logic flow
![image](https://github.com/cisagov/manage.get.gov/assets/141044360/e3c87f4c-b2ae-4586-bd8d-4d72b21c687b)